homeassistant/components/sia/binary_sensor.py

```diff
@@ -86,6 +94,24 @@ def generate_binary_sensors(entry) -> Iterable[SIABinarySensor]:
     For each Zone in each Account there is one smoke and one moisture sensor.
     """
     for account_data in entry.data[CONF_ACCOUNTS]:
+        yield SIABinarySensorConnectivity(
+            port=entry.data[CONF_PORT],
+            account=account_data[CONF_ACCOUNT],
+            zone=SIA_HUB_ZONE,
+            ping_interval=account_data[CONF_PING_INTERVAL],
+            entity_description=ENTITY_DESCRIPTION_CONNECTIVITY,
+            unique_id=SIA_UNIQUE_ID_FORMAT_BINARY.format(
+                entry.entry_id,
+                account_data[CONF_ACCOUNT],
+                SIA_HUB_ZONE,
+                ENTITY_DESCRIPTION_CONNECTIVITY.key,
+            ),
```

Comment:
Looks like we can abstract the unique id for all entities here to use entry_id, accunt, zone and description key. So we can move the definition to a common place.

Reply:
moved all the logic for unique_id and name into a function, without making changes to the result, so non-breaking